rust/src/applayer.rs

```diff
@@ -300,6 +300,10 @@ pub unsafe fn AppLayerRegisterParser(parser: *const RustParser, alproto: AppProt
 
 // Defined in app-layer-detect-proto.h
 extern {
+    pub fn AppLayerProtoDetectPMRegisterPatternCSwPP(iproto: u8, alproto: AppProto,
```

Comment:
Is it worth to get this into Rust ?
All the other protocols do pattern registration from C as far as I know...

Reply:
As a part of this issue, we are trying to get rid of all the `app-layer-<proto>.c/h` files since the entire registration is moving to Rust (as is done in the newer proto parsers). Do you think this is not the right thing to do?

Reply:
ok, I had not understood this goal